### ISSUE TITLE
404 to PATCH contradict RFC 7231

### ISSUE THREAD

**Issue by vfaronov** [json-api/json-api#998]
The “Updating Resources” section of the spec [says](http://jsonapi.org/format/upcoming/#crud-updating-responses-404):

> A server **MUST** return `404 Not Found` when processing a request that references a related resource that does not exist.

This contradicts [RFC 7231 § 6.5.4](http://tools.ietf.org/html/rfc7231#section-6.5.4), which clearly refers to the _target resource_ (i.e. the request URI).

If that’s OK by your compatibility policy, I suggest removing or relaxing this requirement of JSON API. If not, then at least a note is in order.

A better status code may be [422 (Unprocessable Entity)](http://tools.ietf.org/html/rfc4918#section-11.2), defined by WebDAV but widely used elsewhere.

**Comment by ethanresnick** [comment]
+1. 400 or 422 makes more sense.

@json-api/owners?

**Comment by jelhan** [comment]
I fully agree. But I'm not sure if that could be solved without breaking backward compatibility. :cry:
